From: Trond Myklebust <Trond.Myklebust@netapp.com>
Subject: NFS: Fix a bug in nfs_fscache_release_page()
References: bnc#569687
Patch-mainline: 2.6.33-rc7
Git-commit: 2c1740098c708b465e87637b237feb2fd98f129a

    Not having an fscache cookie is perfectly valid if the user didn't mount
    with the fscache option.
    
    This patch fixes http://bugzilla.kernel.org/show_bug.cgi?id=15234
    
Signed-off-by: Trond Myklebust <Trond.Myklebust@netapp.com>
Acked-by: David Howells <dhowells@redhat.com>
Cc: stable@kernel.org
Signed-off-by: Suresh Jayaraman <sjayaraman@suse.de>
---
 fs/nfs/fscache.c |    9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

Index: linux-2.6.32-SLE11-SP1/fs/nfs/fscache.c
===================================================================
--- linux-2.6.32-SLE11-SP1.orig/fs/nfs/fscache.c
+++ linux-2.6.32-SLE11-SP1/fs/nfs/fscache.c
@@ -354,12 +354,11 @@ void nfs_fscache_reset_inode_cookie(stru
  */
 int nfs_fscache_release_page(struct page *page, gfp_t gfp)
 {
-	struct nfs_inode *nfsi = NFS_I(page->mapping->host);
-	struct fscache_cookie *cookie = nfsi->fscache;
-
-	BUG_ON(!cookie);
-
 	if (PageFsCache(page)) {
+		struct nfs_inode *nfsi = NFS_I(page->mapping->host);
+		struct fscache_cookie *cookie = nfsi->fscache;
+
+		BUG_ON(!cookie);
 		dfprintk(FSCACHE, "NFS: fscache releasepage (0x%p/0x%p/0x%p)\n",
 			 cookie, page, nfsi);
 
